File: app/src/main/java/com/blues/framework/widget/endlessbannerview/hintview/ColorPointHintView.java
package com.blues.framework.widget.endlessbannerview.hintview;

import android.content.Context;
import android.graphics.drawable.Drawable;
import android.graphics.drawable.GradientDrawable;


/**
 * Created by Mr.Jude on 2016/1/10.
 */
public class ColorPointHintView extends ShapeHintView {
    private int focusColor;
    private int normalColor;

    public ColorPointHintView(Context context, int focusColor, int normalColor) {
        super(context);
        this.focusColor = focusColor;
        this.normalColor = normalColor;
    }

    @Override
    public Drawable makeFocusDrawable() {
        GradientDrawable dot_focus = new GradientDrawable();
        dot_focus.setColor(focusColor);
        dot_focus.setCornerRadius(dip2px(getContext(), 4));
        dot_focus.setSize(dip2px(getContext(), 8), dip2px(getContext(), 8));
        return dot_focus;
    }

    @Override
    public Drawable makeNormalDrawable() {
        GradientDrawable dot_normal = new GradientDrawable();
        dot_normal.setColor(normalColor);
        dot_normal.setCornerRadius(dip2px(getContext(), 4));
        dot_normal.setSize(dip2px(getContext(), 8), dip2px(getContext(), 8));
        return dot_normal;
    }

    private int dip2px(Context ctx, float dpValue) {
        final float scale = ctx.getResources().getDisplayMetrics().density;
        return (int) (dpValue * scale + 0.5f);
    }
}

File: app/src/main/java/com/blues/framework/widget/endlessbannerview/hintview/ColorPointHintView.kt
package com.blues.framework.widget.endlessbannerview.hintview

import android.content.Context
import android.graphics.drawable.Drawable
import android.graphics.drawable.GradientDrawable
import com.blues.framework.utils.dp

/**
 * Created by Mr.Jude on 2016/1/10.
 */
class ColorPointHintView(context: Context?, private val focusColor: Int,
    private val normalColor: Int) : ShapeHintView(context) {

    override fun makeFocusDrawable(): Drawable {
        return GradientDrawable().apply {
            setColor(focusColor)
            cornerRadius = 4f.dp
            setSize(8f.dp.toInt(), 8f.dp.toInt())
        }
    }

    override fun makeNormalDrawable(): Drawable {
        return GradientDrawable().apply {
            setColor(normalColor)
            cornerRadius = 4f.dp
            setSize(8f.dp.toInt(), 8f.dp.toInt())
        }
    }
}